packages/@webex/plugin-meetings/src/meetings/index.ts

```diff
@@ -813,24 +834,36 @@ export default class Meetings extends WebexPlugin {
     }
 
     return Promise.all([
-      this.fetchUserPreferredWebexSite(),
-      this.getGeoHint(),
-      this.startReachability('registration').catch((error) => {
-        LoggerProxy.logger.error(`Meetings:index#register --> GDM error, ${error.message}`);
-      }),
-      // @ts-ignore
-      this.webex.internal.device
-        .register(deviceRegistrationOptions)
-        // @ts-ignore
-        .then(() =>
-          LoggerProxy.logger.info(
+      this.executeRegistrationStep(() => this.fetchUserPreferredWebexSite(), 'fetchWebexSite'),
+      this.executeRegistrationStep(() => this.getGeoHint(), 'getGeoHint'),
+      this.executeRegistrationStep(
+        () =>
+          this.startReachability('registration').catch((error) => {
+            LoggerProxy.logger.error(`Meetings:index#register --> GDM error, ${error.message}`);
+          }),
+        'startReachability'
+      ),
+      this.executeRegistrationStep(
+        () =>
+          // @ts-ignore
+          this.webex.internal.device
+            .register(deviceRegistrationOptions)
             // @ts-ignore
-            `Meetings:index#register --> INFO, Device registered ${this.webex.internal.device.url}`
-          )
-        )
+            .then(() => {
+              LoggerProxy.logger.info(
+                // @ts-ignore
+                `Meetings:index#register --> INFO, Device registered ${this.webex.internal.device.url}`
+              );
+            }),
+        'deviceRegister'
+      ).then(
         // @ts-ignore
-        .then(() => this.webex.internal.mercury.connect()),
-      MeetingsUtil.checkH264Support.call(this),
+        this.executeRegistrationStep(() => this.webex.internal.mercury.connect(), 'mercuryConnect')
+      ),
+      this.executeRegistrationStep(
+        () => Promise.resolve(MeetingsUtil.checkH264Support.call(this)),
```

Comment:
checkH264Support returns an async function, not a promise so I wrapped it in one (to call .then() on it).